include/seastar/kafka/producer/kafka_producer.hh

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#pragma once
+
+#include <string>
+
+#include "../../../../src/kafka/connection/connection_manager.hh"
+#include "../../../../src/kafka/utils/partitioner.hh"
+#include "../../../../src/kafka/producer/metadata_manager.hh"
+#include "../../../../src/kafka/producer/batcher.hh"
+
+#include <seastar/core/future.hh>
+#include <seastar/net/net.hh>
+#include <seastar/kafka/producer/producer_properties.hh>
+
+namespace seastar {
+
+namespace kafka {
+
+class kafka_producer {
+private:
+
+    producer_properties _properties;
+    lw_shared_ptr<connection_manager> _connection_manager;
+    lw_shared_ptr<metadata_manager> _metadata_manager;
+    batcher _batcher;
+
+public:
+    explicit kafka_producer(producer_properties&& properties);
+    seastar::future<> init();
+    seastar::future<> produce(std::string topic_name, std::string key, std::string value);
```

Comment:
I'm wondering whether string is the right type here for key and value. Wouldn't bytes be a more suitable type?